console/v1/types_console_plugin.go

```diff
@@ -48,6 +48,90 @@ type ConsolePluginSpec struct {
+	//
+	// The OpenShift web console server aggregates the CSP directives and values across
+	// its own default values and all enabled ConsolePlugin CRs, merging them to set a unified
+	// CSP header.
+	//
+	// Example:
+	//   ConsolePlugin A directives:
+	//     script-src: https://script1.com/ https://script2.com/
+	//     font-src: https://font1.com/
+	//
+	//   ConsolePlugin B directives:
+	//     script-src: https://script2.com/ https://script3.com/
+	//     font-src: https://font2.com/
+	//     img-src: https://img1.com/
+	//
+	//   OpenShift web console server CSP response header:
+	//     script-src: self https://script1.com/ https://script2.com/ https://script3.com/
+	//     font-src: self https://font1.com/ https://font2.com/
+	//     img-src: self https://img1.com/
+	//
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=5
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	ContentSecurityPolicy []ConsolePluginCSP `json:"contentSecurityPolicy,omitempty"`
+}
+
+// DirectiveType is an enumeration of OpenShift web console supported CSP directives.
+// LoadType is an enumeration of i18n loading types.
+// +kubebuilder:validation:Enum:=default-src;script-src;img-src;style-src;font-src
+type DirectiveType string
+
+const (
+	// default-src directive serves as a fallback for the other CSP fetch directives.
+	// For more information about the default-src directive, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/default-src
+	DefaultSrc DirectiveType = "default-src"
+	// script-src directive specifies valid sources for JavaScript.
+	// For more information about the script-src directive, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src
+	ScriptSrc DirectiveType = "script-src"
+	// img-src directive specifies a valid sources of images and favicons.
+	// For more information about the img-src directive, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/img-src
+	ImgSrc DirectiveType = "img-src"
+	// style-src directive specifies valid sources for stylesheets.
+	// For more information about the style-src directive, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/style-src
+	StyleSrc DirectiveType = "style-src"
+	// font-src directive specifies valid sources for fonts loaded using @font-face.
+	// For more information about the font-src directive, see:
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/font-src
+	FontSrc DirectiveType = "font-src"
+)
+
+// ConsolePluginCSP holds configuration for a specific CSP directive
+type ConsolePluginCSP struct {
+	// directive specifies which Content-Security-Policy directive to configure.
+	// Available directive types are default-src, script-src, img-src, style-src and font-src.
```

Comment:
Can you expand the godoc here briefly to explain what the directives do, in case the user hasn't explained the higher level field

Eg what is a default vs a script src?

Reply:
Those are explained as part of the enum definition, together with the link for official docs. Thought that it might be more suitable to explain them briefly there together with link.

Reply:
The enum definition won't be generated into docs, please copy to here and make sure that it ends up in the CRD schema for `oc explain` use later